smurfs/src/components/App.js

```diff
@@ -1,22 +1,90 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
 import './App.css';
-/*
- to wire this component up you're going to need a few things.
- I'll let you do this part on your own. 
- Just remember, `how do I `connect` my components to redux?`
- `How do I ensure that my component links the state to props?`
- */
+import { fetchSmurfsStart } from '../actions/index'
+import { smurfSmurf } from '../actions/index'
+
 class App extends Component {
+  state={
+    smurf: {
+      name: '',
+      age: '',
+      height: '',
+    }
+  }
+  componentDidMount(){
+    this.props.fetchSmurfsStart()
+  }
+  handleChange = e => {
+    e.preventDefault()
+    this.setState({
+      smurf: {
+        ...this.state.smurf,
+        [e.target.name]: e.target.value,
+      }
+    })
+  }
+  smurfSmurf = e => {
+    this.props.smurfSmurf(this.state.smurf)
+    this.setState({
+      smurf: {
+        name: '',
+        age: '',
+        height: '',
+      }
+    })
+  }
   render() {
     return (
       <div className="App">
         <h1>SMURFS! 2.0 W/ Redux</h1>
-        <div>Welcome to your Redux version of Smurfs!</div>
-        <div>Start inside of your `src/index.js` file!</div>
-        <div>Have fun!</div>
+          {this.props.smurfs.map((x, i) => (
+            <div  className="smurfs" key={x.name}>
+              <h4>{x.name}</h4>
+              <p>{x.age}</p>
+              <p>{x.height}</p>
+            </div>
+          ))}
+            <form onSubmit={this.smurfSmurf}>
+              <input 
+              type="string"
+              name="name"
+              value={this.state.smurf.name}
+              placeholder="ex, 'Papa Smurf'"
+              onChange={this.handleChange}
+              />
+              <input 
+              type="number"
+              name="age"
+              value={this.state.smurf.age}
+              placeholder="ex, '100'"
+              onChange={this.handleChange}
+              />
+              <input 
+              type="string"
+              name="height"
+              value={this.state.smurf.height}
+              placeholder="ex, '8cm'"
```

Comment:
I really like your placeholders. It's a smart way for the user to know what to write!